### PR TITLE
#22033 Create util method to remove old Time Machine backups files

### PR DESCRIPTION
**Pull request by freddyDOTCMS**
This is a first PR for https://github.com/dotCMS/core/issues/22033

Proposed Changes
Create a Util method to prune old time TimeMachine folder

https://github.com/dotCMS/core/pull/28530/files#diff-5d7a647e09c3a207a6c502bdb945879b865d3dc8ad0cd7cc8c3b4fa233edd3aeR36

Checklist
 Tests
Additional Info
Later, I am going to create a Job to call this method every week

### CODE REVIEW

**Comment by freddyDOTCMS** [comment]
PR: https://github.com/dotCMS/core/pull/28540

File: dotCMS/src/main/java/com/dotcms/timemachine/business/TimeMachineAPIImpl.java [hunk 1]
```diff
@@ -197,4 +203,32 @@ public boolean accept(File dir, String name) {
...
+                    fileToRemove.add(file);
+                }
+            }
+        }
+
+        for (File file : fileToRemove) {
+            try {
+                FileUtil.deleteDir(file.toPath().toString());
+            } catch (IOException e) {
+                //ignore
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/timemachine/business/TimeMachineAPIImpl.java:227]: We should log something here indicating the file that was unable to being removed

File: dotcms-integration/src/test/java/com/dotcms/timemachine/business/TimeMachineAPITest.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                    45, 89);
+
+            final List<File> removedFiles = APILocator.getTimeMachineAPI().removeOldTimeMachineBackupsFiles();
+
+            assertEquals(expireFolders.size(), removedFiles.size());
+            assertTrue(removeAll(expireFolders, removedFiles).isEmpty());
+
+            final File[] files = tmTestingFolder.listFiles();
+            assertEquals(notExpireFolders.size(), files.length);
+            assertTrue(removeAll(Arrays.asList(), notExpireFolders).isEmpty());
```
Comment (dsilvam) [inline dotcms-integration/src/test/java/com/dotcms/timemachine/business/TimeMachineAPITest.java:55]: this looks incomplete: "Arrays.asList()"

File: dotcms-integration/src/test/java/com/dotcms/timemachine/business/TimeMachineAPITest.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        final String timemachinePathPreviousValue = Config.getStringProperty("TIMEMACHINE_PATH", null);
+
+        try {
+            final List<File> expireFolders = createFiles(tmTestingFolder, "", 90, 100);
+            final List<File> removedFiles = APILocator.getTimeMachineAPI().removeOldTimeMachineBackupsFiles();
+            assertTrue(removedFiles.isEmpty());
+
+            final File[] files = tmTestingFolder.listFiles();
+            assertEquals(expireFolders.size(), files.length);
+            assertTrue(removeAll(Arrays.asList(), expireFolders).isEmpty());
```
Comment (dsilvam) [inline dotcms-integration/src/test/java/com/dotcms/timemachine/business/TimeMachineAPITest.java:125]: Incomplete also: "Arrays.asList()"

File: dotCMS/src/main/java/com/dotcms/timemachine/business/TimeMachineAPIImpl.java (hunk 1 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/timemachine/business/TimeMachineAPIImpl.java:227]: agree

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/timemachine/business/TimeMachineAPIImpl.java (hunk 1 above)
Comment (freddyDOTCMS) [reply dotCMS/src/main/java/com/dotcms/timemachine/business/TimeMachineAPIImpl.java]: done https://github.com/dotCMS/core/pull/28530/commits/6e6757d800ae36575a4fd3854c1a6e164d8bd327#diff-19b28e2982b6db869240fb8fc96a2bf2c156e3607f41656fd16d7845c45ba604R228

File: dotcms-integration/src/test/java/com/dotcms/timemachine/business/TimeMachineAPITest.java (hunk 2 above)
Comment (freddyDOTCMS) [reply dotcms-integration/src/test/java/com/dotcms/timemachine/business/TimeMachineAPITest.java]: done https://github.com/dotCMS/core/pull/28530/commits/6e6757d800ae36575a4fd3854c1a6e164d8bd327#diff-f27467cb9919462f8c7c5a137c3c9b9bf21be349047c0d1e1ba9315773e53298R55

File: dotcms-integration/src/test/java/com/dotcms/timemachine/business/TimeMachineAPITest.java (hunk 3 above)
Comment (freddyDOTCMS) [reply dotcms-integration/src/test/java/com/dotcms/timemachine/business/TimeMachineAPITest.java]: done https://github.com/dotCMS/core/pull/28530/commits/6e6757d800ae36575a4fd3854c1a6e164d8bd327#diff-f27467cb9919462f8c7c5a137c3c9b9bf21be349047c0d1e1ba9315773e53298R125

**Review by dsilvam — APPROVED** [review]

**Pull request merged**